1. Basic Web Development/frontend/src/main.ts

```diff
@@ -0,0 +1,419 @@
+
+            document.getElementById(elemenToEdit.type + 'Edit').click();
+            (document.getElementById('permNameEdit') as HTMLInputElement).value = elemenToEdit.text;
+
+            createPermissionsForm(form);
+        } else if (target.classList.contains('delete-btn')) {
+            if(NOWEDITING.id == target.name) {
+                document.getElementById('edit-form').classList.add('no-display');
+            }
+
+            this.PERMISSIONS = this.PERMISSIONS.filter((el: Permission) => el.id != target.name);
+
+            createPermissionsForm(form);
+
+            fetch(API_URL + '/pref/delete', { method: 'DELETE', headers: HEADERS, body: JSON.stringify({ name: target.name }) })
+                .then(response => response.json())
+                .catch((error) => {
+                    console.error('Error:', error);
+                    permissionsRoutine();
+                });
+        }
+    });
+}
+
+function createPermissionsForm(form: HTMLElement) {
+    const permissionsList = document.getElementById("permissions-list");
+
+    removeChildren(permissionsList);
+
+    this.PERMISSIONS.forEach((element: Permission) => {
+        const newElement = document.createElement("div");
+        newElement.classList.add("grid-element");
+
+        const newLabel = document.createElement("label");
+        newLabel.setAttribute('for', element.id);
+        newLabel.innerHTML = element.text;
+
+        const newInput = document.createElement("input");
+        newInput.setAttribute('id', element.id);
+        newInput.setAttribute('name', element.type);
+        newInput.setAttribute('type', 'checkbox');
+
+        if (element.type == "permissionNss") {
+            newInput.click();
+            newInput.disabled = true;
+        }
+
+        const newButtonDelete = document.createElement("button");
+        newButtonDelete.setAttribute('type', 'button');
+        newButtonDelete.textContent = 'X';
+        newButtonDelete.setAttribute('name', element.id);
+        newButtonDelete.classList.add('letter-btn');
+        newButtonDelete.classList.add('delete-btn');
+
+        const newButtonEdit = document.createElement("button");
+        newButtonEdit.setAttribute('type', 'button');
+        newButtonEdit.textContent = 'E';
+        newButtonEdit.setAttribute('name', element.id);
+        newButtonEdit.classList.add('letter-btn');
+        newButtonEdit.classList.add('edit-btn');
+
+        newElement.appendChild(newButtonEdit);
+        newElement.appendChild(newButtonDelete);
+        newElement.appendChild(newLabel);
+        newElement.appendChild(newInput);
+
+        permissionsList.appendChild(newElement);
+    });
+
+    setCounter();
+
+    this.BOXES = getBoxes();
+}
+
+function setCounter() {
+    const counterDisplay = document.getElementById("counter-display");
+    removeChildren(counterDisplay);
+
+    this.COUNTER = document.querySelectorAll('input[type="checkbox"]:checked').length;
```

Comment:
Is this selector specific enough? 

![image](https://user-images.githubusercontent.com/17475808/126177813-dc043f5e-7727-473b-b109-a14146bf7c9b.png)